src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/readers/JSONTemplateReader.java

```diff
@@ -144,9 +143,11 @@ private void getBuilderFromJsonObject(
                     } else if (valueNode.isArray()) {
                         getBuilderFromJsonArray(entryName, valueNode, currentBuilder, maker);
                     } else {
-                        if (!jumpField)
+                        if (!jumpField) {
```

Comment:
put in a separate commit.
The fix was meant to allow the user to not have to strictly specify the top level feature source since can be automatically inferred from GeoServer.